core/src/main/scala/scalafix/util/ChangeType.scala

```diff
@@ -0,0 +1,43 @@
+package scalafix.util
+
+import scala.collection.immutable.Seq
+import scala.{meta => m}
+import scalafix.rewrite._
+
+//Provide a little structure to the replacements we will be performing
+case class ReplaceType(original: m.Type,
+                       replacement: m.Type,
+                       newString: String) {
+  def toPatch(t: m.Type): Patch = Patch.replace(t, newString)
+}
+
+class ChangeType(ast: m.Tree, ctx: RewriteCtx)(implicit sApi: SemanticApi) {
+  object DType extends Desugared[m.Type]
+  object DTerm extends Desugared[m.Term]
+
+  def partialTypeMatch(
+      rt: ReplaceType): PartialFunction[m.Tree, (m.Type, ReplaceType)] = {
+    case t @ DType(desugared)
+        if StructurallyEqual(desugared, rt.original).isRight =>
+      t -> rt
+  }
+
+  def partialTypeMatches(replacementTypes: Seq[ReplaceType])
+    : PartialFunction[m.Tree, (m.Type, ReplaceType)] =
+    replacementTypes.map(partialTypeMatch).reduce(_ orElse _)
+
+  // NOTE. This approach is super inefficient, since we run semantic.desugar on
```

Comment:
Ditto, this is really inefficient. But I say we don't worry about it until it's a problem.